pages/querying/time-to-live.mdx

```diff
@@ -0,0 +1,148 @@
+---
+title: Time to live
+description: Learn how to use time-to-live in Memgraph.
+---
+
+import { Callout } from 'nextra/components'
+
+<Callout type="info">
+
+This is an ENTERPRISE feature.
+
+</Callout>
+
+# Time to live (Enterprise)
+
+Time-to-live allows a user to tag vertices with an expiration time. Once a vertex has expired, the vertex and all associated edges will be deleted.
+
+<Callout type="info">
+
+Time-to-live is implemented as a background job that periodically gets executed. 
+This is a best effort solution; meaning that even if a vertex expires, that does not mean it gets deleted right away, but eventually, once the background job gets executed.
+
+</Callout>
+
+## Usage
+
+In order to use the feature the user needs to:
+1. [Configure TTL](#configuration)
+2. [Tag vertices with an expiration time](#tagging-vertices)
+
+Once that is done, a background job will periodically delete expired vertices and associated edges.
+
+### What is indexed
+
+Time-to-live uses a label `TTL` and property `ttl` to tag vertices. A label+property value index is used to speed up query execution.
+
+### Executed query
+
+Time-to-live is implemented as a background job that execute the following query:
+
+```cypher
+MATCH (n:TTL) WHERE n.ttl < $now WITH n LIMIT $batch DETACH DELETE n;
+```
+
+The query DETACH DELETEs all vertices that have expired at this point in time.
+The query is batched to limit the serialization errors and lost work that the error might cause.
+
+<Callout type="warning">
+
+Since time-to-live is implemented as a query like any other, the user might get serialization errors.
+This can happen if the user is modifying an expired vertex.
+The chance of serialization errors can be minimized by limiting the duration of write transactions.
+In addition, the user can disable TTL before starting an important write transaction and re-enable it after commit. See [configuration](#configuration).
+
+</Callout>
+
+## Configuration
+
+Time to live is by default disabled. To enable it, run:
+
+```cypher
+ENABLE TTL [EVERY "period"] [AT "time"];
+```
+
+ * period: a string literal defining the period of execution. Format: "NdNhNmNs"
+ * time: a specific time at which the ttl job gets executed. Format: "24:59:59"
```

Comment:
I'm assuming this is the FIRST time that the ttl job runs. After that, the ttl timing is controlled by the "period". Please add an example.